test/e2e/marketplace_test.go

```diff
@@ -0,0 +1,186 @@
+		},
+	}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, operatorsourceList)
+	if err != nil {
+		t.Fatalf("failed to add operatorsource custom resource scheme to framework: %v", err)
+	}
+	err = framework.AddToFrameworkScheme(apis.AddToScheme, catalogsourceconfigList)
+	if err != nil {
+		t.Fatalf("failed to add catalogsourceconfig custom resource scheme to framework: %v", err)
+	}
+	// run subtests
+	t.Run("marketplace-group", func(t *testing.T) {
+		t.Run("Cluster", MarketplaceCluster)
+	})
+}
+
+func defaultCreateTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("could not get namespace: %v", err)
+	}
+
+	testOperatorSource := &operator.OperatorSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "global-operators",
+			Namespace: namespace,
+		},
+		Spec: operator.OperatorSourceSpec{
+			Type:              "appregistry",
+			Endpoint:          "https://quay.io/cnr",
+			RegistryNamespace: "operators",
+		},
+	}
+
+	catalogSourceConfigName := "opsrc-global-operators"
+	configMapName := "csc-cm-opsrc-global-operators"
+	catalogSourceName := "csc-cs-opsrc-global-operators"
+
+	// Create the operatorsource to download the manifests.
+	err = f.Client.Create(
+		context.TODO(),
+		testOperatorSource,
+		&framework.CleanupOptions{
+			TestContext:   ctx,
+			Timeout:       cleanupTimeout,
+			RetryInterval: cleanupRetryInterval,
+		})
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig.
+	testCatalogSourceConfig := &operator.CatalogSourceConfig{}
+	err = WaitForResult(t, f, testCatalogSourceConfig, namespace, catalogSourceConfigName)
+	if err != nil {
+		return err
+	}
+
+	// Check for the config map created from the catalogsourceconfig.
+	err = WaitForResult(t, f, &corev1.ConfigMap{}, namespace, configMapName)
+	if err != nil {
+		return err
+	}
+
+	// Then check for the catalog source.
+	err = WaitForResult(t, f, &olm.CatalogSource{}, namespace, catalogSourceName)
+	if err != nil {
+		return err
+	}
+
+	// Now lets update the catalogsourceconfig and see that the configmap and catalog source are updated
+	// First set the target namespace field for the catalogsourceconfig to a new namespace
+	testCatalogSourceConfig.Spec.TargetNamespace = testNamespace
+	err = f.Client.Update(context.TODO(), testCatalogSourceConfig)
+	if err != nil {
+		return err
+	}
+
+	// Check for the config map created in the new namespace
+	err = WaitForResult(t, f, &corev1.ConfigMap{}, testNamespace, configMapName)
+	if err != nil {
+		return err
+	}
+
+	// Then check for the catalog source in the new namespace
+	err = WaitForResult(t, f, &olm.CatalogSource{}, testNamespace, catalogSourceName)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func MarketplaceCluster(t *testing.T) {
+	t.Parallel()
```

Comment:
What does this parallelize?

Reply:
Right now nothing because the test is only called once. It's a flag for test framework that, if other tests are called at the same time, they will be run in parallel rather than serially.

Reply:
Do we want this behavior by default? What if we introduce an update CatalogSourceConfig test case that depends on the OperatorSource / CatalogSourceConfig created here to be present? I suggest we introduce this when we really have tests that we want to run in parallel.